application-analytics-ui/src/main/resources/Analytics/Code/AddTrackingCode.xml

```diff
@@ -169,6 +169,7 @@
 #set ($enabled = $configuration.enabled)
 #if ($trackingCode &amp;&amp; $enabled)
   {{html clean="false"}}
+  &lt;meta name='analyticsProUserReference' content = "$crtUserDoc.getDocumentReference()" /&gt;
```

Comment:
How is the `analyticsProUserReference` metadata actually used? Is it defined in matomo?

Reply:
The <html> tag seems to have an attribute that points towards the current user `data-xwiki-user-reference`. I think you could try to use that. However, I couldn't find anything documenting it. I reckon thats where the javascript API looks when retrieving the current user. WDYT @oanalavinia @Farcasut 

![Uploading image.png…]()

Reply:
Hi! analyticsProUserReference is just an arbitrary name made specifically for the analytics app and is not defined by Matomo. I chose this name to ensure it is unique and doesn't conflict with other names. The metadata is used if you want to make user tracking non-anonymous, i.e., assign an ID to each visit corresponding to the user. When adding the tracking code to your instance you need to append the user id to the tracking code:

```html
<!-- Matomo -->
<script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  _paq.push(['setUserId', document.querySelector('meta[name="analyticsProUserReference"]').attributes.content.textContent]);
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://piwik.xwiki.com/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '34']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->

```

Reply:
Indeed, there is a `user-reference` added always in the [html header](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm#L61) , and you could reuse it

Reply:
The guest users can be treated something like this:
```html
<!-- Matomo -->
<script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  let userReference = document.querySelector('html').getAttribute('data-xwiki-user-reference');
  let id = userReference != null ? userReference : 'guest';
  _paq.push(['setUserId', id ]);
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://piwik.xwiki.com/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '34']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->
```